mobile/test/swift/integration/proxying/HTTPRequestUsingProxyTest.swift

```diff
@@ -0,0 +1,124 @@
+import Envoy
+import EnvoyEngine
+import EnvoyTestApi
+import EnvoyTestServer
+import Foundation
+import TestExtensions
+import XCTest
+
+final class HTTPRequestUsingProxyTest: XCTestCase {
+  override static func setUp() {
+    super.setUp()
+    register_test_extensions()
+  }
+
+  func testHTTPRequestUsingProxy() throws {
+    EnvoyTestServer.startHttpProxyServer()
+    let port = EnvoyTestServer.getEnvoyPort()
+
+    let engineExpectation = self.expectation(description: "Run started engine")
+    let responseHeadersExpectation =
+        self.expectation(description: "Successful response headers received")
+    let responseTrailersExpectation =
+        self.expectation(description: "Successful response trailers received")
+
+    let engine = EngineBuilder()
+      .addLogLevel(.trace)
+      .setOnEngineRunning {
+        engineExpectation.fulfill()
+      }
+      .enableProxying(true)
+      .build()
+
+    EnvoyTestApi.registerTestProxyResolver("127.0.0.1", port: port)
+
+    XCTAssertEqual(XCTWaiter.wait(for: [engineExpectation], timeout: 5), .completed)
+
+    // Send a request to trigger the test filter which should log an event.
+    let requestHeaders = RequestHeadersBuilder(method: .get, scheme: "http",
+                                               authority: "neverssl.com", path: "/")
```

Comment:
Note to self: change this to use a FakeUpstream instead of an actual destination on the web.

Reply:
Will do this in a follow up PR (the Android tests also use actual web destinations for these tests)